src/Factory/OccurrencesFactory.php

```diff
@@ -0,0 +1,109 @@
+use App\Entity\Occurrence;
+use App\Model\Feed\FeedItemOccurrence;
+use App\Repository\OccurrenceRepository;
+use Psr\Log\LoggerInterface;
+
+final class OccurrencesFactory
+{
+    public function __construct(
+        private readonly OccurrenceRepository $occurrenceRepository,
+        private readonly LoggerInterface $logger
+    ) {
+    }
+
+    /**
+     * Create occurrences or find matching in the database.
+     *
+     * @param array<FeedItemOccurrence> $input
+     *   The tag names to create/lockup in the database as strings
+     * @param Event $event
+     *   The event related to the occurrences
+     */
+    public function createOrLookup(array $input, Event $event): void
+    {
+        $eventOccurrences = $event->getOccurrences();
+        foreach ($eventOccurrences as $occurrence) {
+            foreach ($input as $id => $item) {
+                // Check if item exist in the old occurrences base on timestamps.
+                if ($this->isEqualDates($item, $occurrence, $event)) {
+                    // Update with new values.
+                    $this->setValues($item, $occurrence);
+                    $this->occurrenceRepository->save($occurrence);
+
+                    // Removed processed occurrence from input.
+                    unset($input[$id]);
+
+                    // Jump to outer foreach.
+                    continue 2;
+                }
+            }
+
+            // Not found in input, so remove it.
+            $event->getOccurrences()->removeElement($occurrence);
+        }
+
+        // Loop over remaining input elements.
+        foreach ($input as $item) {
+            $occurrence = new Occurrence();
+            $this->setValues($item, $occurrence);
+            $this->occurrenceRepository->save($occurrence);
+
+            // Add the new occurrence to the event.
+            $event->addOccurrence($occurrence);
+        }
+    }
+
+    /**
+     * Compare dates for an occurrence.
+     *
+     * @param FeedItemOccurrence $item
+     *   The new input occurrence
+     * @param occurrence $occurrence
+     *   An occurrences from an exiting database backed event
+     * @param Event $event
+     *   The event in question
+     *
+     * @return bool
+     *   True if dates are equal else false
+     */
+    private function isEqualDates(FeedItemOccurrence $item, Occurrence $occurrence, Event $event): bool
+    {
+        $occurrenceStartDate = $occurrence->getStart();
+        $occurrenceEndData = $occurrence->getEnd();
+        if (!isset($occurrenceStartDate, $occurrenceEndData, $item->start, $item->end)) {
+            // This should not happen.
+            $this->logger->critical(sprintf('Event (id: %d) has occurrences dates that are null', $event->getId() ?? '-1'));
+
+            // We return null to remove the event.
+            return false;
+        }
+
+        return 0 === $occurrenceStartDate->diff($item->start)->s && 0 === $occurrenceEndData->diff($item->end)->s;
```

Comment:
There's no need for creating `DateInterval` instances to check if dates are equal:

```suggestion
        return $occurrenceStartDate->getTimestamp() === $item->start->getTimestamp()
          && $occurrenceEndDate->getTimestamp() === $item->end->getTimestamp();
```